enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+    // No Cloud Controller Manager is expected to be supplied.
+    // Signaling MCO not to set `--cloud-provider` flag to the kubelets.
+    CloudControllerManagerDisabled CloudControllerManagerMode = "Disabled"
+)
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct{
+    // CloudControllerManager determines whether or not an external Cloud Controller Manager is expected to 
+    // be presented in the cluster.
+    // For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When enabled, the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+    // When omitted or disabled, no `cloud-provider` flag should be set.
+    // +kubebuilder:validation:Enum=Enabled;Disabled
+    // +optional
+    CloudControllerManager CloudControllerManagerMode `json:"cloudControllerManager,omitempty"`
+}
+
+// ExternalPlatformStatus holds the current status of the generic External infrastructure provider.
+type ExternalPlatformStatus struct{}
+```
+
+```go
+type PlatformSpec struct {
+    ...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformSpec `json:"external,omitempty"`
+}
+
+type PlatformStatus struct {
+	...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformStatus `json:"external,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create support burden. However, using of such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signals about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle hardened.
+Changing it with adding new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control. 
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
```

Comment:
New features like this are being introduced as TPNU now, so yes, yes we should do this! David has some WIP to update our generators to support this

Reply:
Is there any enhancement/design document which we can refer there?

Reply:
I know there's https://docs.google.com/document/d/1K2-zWyYQH-tHvPVwcKsPMxPJYNEOH5N3tENsBjDnqI0/edit#heading=h.c97miuz336km, but I don't think it became an enhancement